machineLearning/stockRecommendationSystem/modelGeneration/model_3/model_3.py

```diff
@@ -0,0 +1,216 @@
+# Module imports
+import pandas as pd
+import numpy as np
+import os
+
+from sklearn.metrics.pairwise import cosine_similarity
+from sklearn import preprocessing
+from sentence_transformers import SentenceTransformer
+from pymongo import MongoClient
+from dotenv import load_dotenv
+
+# This function is used to read in the original dataset and save the computed vectors as a pickle file.
+def read_data():
+    # Read in the s&p dataset
+    stocks = pd.read_csv('machineLearning\stockRecommendationSystem\modelGeneration\model_3\stocks.csv')
+
+    # Remove some columns that won't be used
+    stocks.drop(['idnumber','longnamesort', 'weight'], inplace=True, axis=1)
+    # print(stocks)
+    print(stocks.info())
+    # Create array X of the all long business summaries, sectors, industries and countries and exchange
+    X = np.array(stocks.longbusinesssummary + " " + stocks.sector + " " + stocks.industry + " " + stocks.country + " " + stocks.exchange)
+
```

Comment:
Just a thought is the exchange a meaningful input? the snp only has stocks from like 3 exchanges. Also country we only have 19 not from America. Wonder would it cause any issues with overfitting or never include similar stocks to non american stocks if that's the case. Could be wrong just food for thought anyway.

Reply:
It actually doesn't change the output at all because those inputs get diluted down when compared to the longbusinesssummary. But yep can remove them either, my thinking was that we have the data there and it's easy to add to the "bag of words" that are being compared so might as well.